File: quan_ly_tu_do_API_one/quan_ly_tu_do_API/Models/Aisuggestion.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace quan_ly_tu_do_API.Models;

/// <summary>
/// Lưu lịch sử gợi ý trang phục từ AI (HomeTab → suggestOutfit).
/// Một bản ghi = một lần AI đề xuất bộ trang phục cho user.
/// </summary>
[Table("ai_suggestions")]
[Index("UserId", "CreatedAt", Name = "idx_ai_suggestions_user_date", IsDescending = new[] { false, true })]
public partial class AiSuggestion
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("user_id")]
    public Guid UserId { get; set; }

    // ── Ngữ cảnh lúc gợi ý ────────────────────────────────────────────

    /// <summary>Địa điểm / tag đích (vd: "Đi làm", "Đi chơi", "")</summary>
    [Column("destination")]
    [StringLength(150)]
    public string? Destination { get; set; }

    /// <summary>Tag sức khoẻ user chọn (vd: "Khoẻ mạnh", "Mệt mỏi")</summary>
    [Column("health_tag")]
    [StringLength(100)]
    public string? HealthTag { get; set; }

    /// <summary>Snapshot thời tiết lúc gợi ý (vd: "Nắng, 32°C, Độ ẩm 75%")</summary>
    [Column("weather_snapshot")]
    [StringLength(255)]
    public string? WeatherSnapshot { get; set; }

    // ── Kết quả AI trả về ─────────────────────────────────────────────

    /// <summary>Nội dung AI gợi ý (markdown/plain text)</summary>
    [Column("suggestion_text")]
    public string SuggestionText { get; set; } = null!;

    /// <summary>Nguồn gợi ý: "rag" | "worker" | "heuristic"</summary>
    [Column("source")]
    [StringLength(30)]
    public string Source { get; set; } = "worker";

    // ── Phản hồi của user ─────────────────────────────────────────────

    /// <summary>null = chưa phản hồi, true = hữu ích, false = không hữu ích</summary>
    [Column("is_helpful")]
    public bool? IsHelpful { get; set; }

    /// <summary>ID outfit user tạo từ gợi ý này (nếu có)</summary>
    [Column("saved_outfit_id")]
    public Guid? SavedOutfitId { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    // ── Navigation ────────────────────────────────────────────────────
    [ForeignKey("UserId")]
    public virtual User User { get; set; } = null!;

    [ForeignKey("SavedOutfitId")]
    public virtual Outfit? SavedOutfit { get; set; }
}

File: tuquanaoAI/lib/models/ai_suggestion.dart
class AiSuggestion {
  final String id;
  final String userId;
  final String? destination;
  final String? healthTag;
  final String? weatherSnapshot;
  final String suggestionText;
  final String source; // 'rag' | 'worker' | 'heuristic'
  final bool? isHelpful;
  final String? savedOutfitId;
  final DateTime createdAt;

  const AiSuggestion({
    required this.id,
    required this.userId,
    this.destination,
    this.healthTag,
    this.weatherSnapshot,
    required this.suggestionText,
    required this.source,
    this.isHelpful,
    this.savedOutfitId,
    required this.createdAt,
  });

  factory AiSuggestion.fromJson(Map<String, dynamic> json) => AiSuggestion(
    id:              json['id'] as String,
    userId:          json['userId'] as String,
    destination:     json['destination'] as String?,
    healthTag:       json['healthTag'] as String?,
    weatherSnapshot: json['weatherSnapshot'] as String?,
    suggestionText:  json['suggestionText'] as String,
    source:          json['source'] as String? ?? 'worker',
    isHelpful:       json['isHelpful'] as bool?,
    savedOutfitId:   json['savedOutfitId'] as String?,
    createdAt: DateTime.parse(json['createdAt'] as String),
  );

  Map<String, dynamic> toJson() => {
    'id':              id,
    'userId':          userId,
    'destination':     destination,
    'healthTag':       healthTag,
    'weatherSnapshot': weatherSnapshot,
    'suggestionText':  suggestionText,
    'source':          source,
    'isHelpful':       isHelpful,
    'savedOutfitId':   savedOutfitId,
    'createdAt':       createdAt.toIso8601String(),
  };

  AiSuggestion copyWith({bool? isHelpful, String? savedOutfitId}) =>
      AiSuggestion(
        id:              id,
        userId:          userId,
        destination:     destination,
        healthTag:       healthTag,
        weatherSnapshot: weatherSnapshot,
        suggestionText:  suggestionText,
        source:          source,
        isHelpful:       isHelpful ?? this.isHelpful,
        savedOutfitId:   savedOutfitId ?? this.savedOutfitId,
        createdAt:       createdAt,
      );
}